### ISSUE TITLE
Training queries

### ISSUE THREAD

**Issue by srijandas07** [EGO4D/forecasting#8]
I have recently observed that while training the Longterm Action Anticipation model (with pre-trained slowfast weights), the loss indicates nan.
Also May I know the expected training time per epoch? 
<img width="560" alt="Screen Shot 2022-04-28 at 11 08 25 PM" src="https://user-images.githubusercontent.com/24988882/165879884-fe043311-4721-43ed-8f5b-eb05c17d0a48.png">

**Comment by antoninofurnari** [comment]
Hello,

I remember training time being in the order of 30 minutes per epoch, training on 4 V100 GPUs.

Regarding the nan loss, did you change any hyperparameters or the number of GPUs? The nan loss may be due to a high learning rate. As a reference, you expect to see the training loss going down in Tensorboard as shown below:

![image](https://user-images.githubusercontent.com/20006211/165886873-9be5bc68-0fd4-4f78-8456-998dc984e086.png)

**Comment by srijandas07** [comment]
I guess the script is configured for 8 GPUs. However, I am training it on 4 A5000 GPUs.
So, I have just changed the number of GPUs.
Also, at validation we encounter an error at line 291 of ego4d/models/lta_models.py
preds_dist = Categorical(logits=head_x)
<img width="1002" alt="Screen Shot 2022-04-29 at 1 20 46 AM" src="https://user-images.githubusercontent.com/24988882/165888943-74bf55f6-c609-4d61-863a-e4ac20b5c0a5.png">

**Comment by antoninofurnari** [comment]
Apologies! I had mistakenly read "short-term anticipation", so my comment above (which I have now deleted in order to avoid confusion) was related to that. Please forget about what I wrote before.

I guess @Tushar-N can best reply to your questions.

Sorry for the confusion.

Antonino

**Comment by Tushar-N** [comment]
I've tested the script on 2 GPUs and 8 GPUs and didn't encounter any issues with nans. Could you check that the package versions (pytorchvideo, pytorch etc.) are consistent with the requirements.txt file?

Training time for the slowfast concat model should be about 22 hours on 8 GPUs (or about 25 mins per epoch). Here's a snapshot from the tensorboard log files.
<img width="357" alt="Screen Shot 2022-04-29 at 1 32 52 PM" src="https://user-images.githubusercontent.com/4995097/166004474-e4c734ce-68f0-4d0b-9daa-195a8bd808be.png">

**Comment by srijandas07** [comment]
Just a note - I could successfully run the framework with Concat decoder but encounter the nan's while using the Transformer decoder.
I have checked the nan's are coming even at validation sanity check. It might be due to FP16? 
Were you using V100s?
The training time makes sense. For your experiments with 2 GPUs, did you rescaled the learning parameter or other hyper-parameters?
Thanks in advance.

**Comment by srijandas07** [comment]
I realized the difference in configurations. This is solved now.
Thanks
